packages/app/src/specs/SpecsList.vue

```diff
@@ -13,73 +13,83 @@
       :gql="props.gql.currentProject"
       @close="showSpecPatternModal = false"
     />
-
-    <template
+    <!--
+      The markup around the virtualized list is pretty delicate. We might be tempted to
+      combine the `v-if="specs.length"` and the `:class="specs.length ? 'grid': 'hidden'"` below
+      into a single v-if on a `<template>` that would wrap both. This creates a problem
+      recovering from the empty results state as the `.spec-list-container` element is needed
+      in order to render the virtual dom. Virtualized lists require stable containers and it
+      can be tricky to debug after this has happened.
+    -->
+    <div
```

Comment:
How can a conditional with `v-if` be stable in the first place (what does stable mean, exactly?)

If we need a DOM node that will never change, we could use the `v-once` directive.

Reply:
This is a good point, "stable" was vague. I've moved the comment to be better placed, and updated the wording. The part with `v-if` is OK coming and going, it's the next element that shouldn't follow the `v-if` rule. We would have liked to use v-show but it clashed with tailwind's `grid` class.

Maybe it's overdoing it to have this long comment in the first place, I think some "here be dragons" is useful for future developers, but there are tests now to catch regressions of the specific stuff this fixes.

Reply:
I still don't fully understand the problem, I guess I need to pull and try it out - but that's okay, the comment should suffice. Hopefully I don't need to edit this anytime soon.

Reply:
The specific problem that this change fixes is that if the container element is not present in the DOM with when this code runs, it doesn't update the virtualized list, so "clearing the search" from the no-results state was broken, as the old `v-if` had removed the element needed. @ZachJW34 has some ideas for making this more robust in general when we get a chance. But in this PR I just wanted to fix the bug I found when adding the coverage for that behavior, and give a bit of a here-be-dragons, because some of my first changes seemed to work great locally, but had actually broken the virtualization.

https://github.com/cypress-io/cypress/blob/317d7592c01cc9224e6cb82de87f34517792a595/packages/frontend-shared/src/composables/useVirtualList.ts#L84